utils/s2n_random.c

```diff
@@ -512,11 +512,22 @@ int s2n_openssl_compat_status(void)
     return 1;
 }
 
+#if S2N_LIBCRYPTO_SUPPORTS_ENGINE
 int s2n_openssl_compat_init(ENGINE *unused)
 {
     return 1;
 }
 
+/* S2N_LIBCRYPTO_SUPPORTS_ENGINE is disabled when compiling with AWS-LC.
+ *
+ * Custom random is only used when linked with OpenSSL. However, it should be
+ * possible to compile the ENGINE related code with other libcrypto (eg.
+ * BoringSSL, LibreSSL). This is not the case with AWS-LC which has a different
+ * `RAND_METHOD` signature and results in a compilation error.
+ *
+ * [1] AWS-LC: https://github.com/aws/aws-lc/blob/main/include/openssl/rand.h#L124
+ * [2] OpenSSL: https://github.com/openssl/openssl/blob/master/include/openssl/rand.h#L42
+ */
```

Comment:
Nit: This comment is probably overkill / off topic. The only part that's relevant to this piece of code is that RAND_METHOD may not be defined, and that aws-lc does define it but differently.

Reply:
> RAND_METHOD may not be defined

RAND_METHOD is not an ENGINE api so I dont think this part is true. The signature mismatch is the reason for gating this.